docs/native-code.md

```diff
@@ -0,0 +1,127 @@
+
+C# development brings engineering efficiencies to writing a module or an app. We strive to maintain parity in developer experiences between C# and C++. If your app or module already uses C#, you should feel empowered to continue to use C#. That said, modules that Microsoft contributes to will be written in C++ to ensure the highest level of performance and interoperability.
+
+<div class="warning">
+  <h3>Important</h3>
+  <h4>Mixing C# and C++</h4>
+</div>
+
+### App crashes when trying to load the C# component or instantiate one of its types
+**Error 0x80131040 “The located assembly’s manifest definition does not match the assembly reference”**
+
+C++ apps consuming native modules written in C# need special care. There is a bug in the interop between C# and C++: https://github.com/microsoft/dotnet/issues/1196.
+
+The symptoms are that building the app will work fine but the C++ app will crash at runtime when trying to load the C# module with the error `0x80131040 : The located assembly's manifest definition does not match the assembly reference.`
+
+A write-up of the problem can be found [here](https://devblogs.microsoft.com/oldnewthing/20200615-00/?p=103868/). 
+To work around this problem there are three options:
+1. Set your C# component's target Windows version to Windows 10 version 1703 (Build 15063) or lower.
+1. Reference the .NET Native nuget packages in your C++ app:
+   - Right click on the app's .vcxproj file → **Manage NuGet Packages**.
+   - Search for `Microsoft.Net.Native.Compiler`, and install it.
+   - Then add the following properties to the .vcxproj file:
+    ```xml
+    <PropertyGroup>
+      <UseDotNetNativeToolchain Condition="'$(Configuration)'=='Release'">true</UseDotNetNativeToolchain>
+      <DotNetNativeVersion>2.2.3</DotNetNativeVersion>
+    </PropertyGroup>
+    ```
+1. In your .vcxproj file, set this property in the first `<PropertyGroup>`:
+   ```xml
+   <ConsumeCSharpModules>true</ConsumeCSharpModules>
+   ```
+
+## Troubleshooting and debugging C++ native modules
+
+So you added a new native module or a new method to a module but it isn't working, **now what?!**
+
+If your method isn't being hit in the VS debugger, something is blocking the call due to a mismatch, likely between the expected and actual types that your method takes/returns.
+
+To debug into what is rejecting the call, set a breakpoint in `CxxNativeModule::invoke` (See [ReactCommon\react-native-patched\ReactCommon\cxxreact\CxxNativeModule.cpp](https://github.com/facebook/react-native/blob/0b8a82a6eeeb3508b80ee137d313f64fe323db06/ReactCommon/cxxreact/CxxNativeModule.cpp#L97)). This breakpoint is bound to be hit a lot (every time a call to a native method is made), so we want to make sure we only break when *our* method of interest is involved.

```

Comment:
We probably don't want to point folks towards this. react-native-patched is a build time directory we're trying to remove (it will be gone on binary distribution as well), and Facebook has been refactoring some of this.

Reply:
We can update the docs when the file gets removed.